### ISSUE TITLE
Question about "option is deprecated"

### ISSUE THREAD

**Issue by schmunk42** [fxpio/composer-asset-plugin#293]
```
The "extra.asset-installer-paths" option is deprecated, use the "config.fxp-asset.installer-paths" option
The "extra.asset-pattern-skip-version" option is deprecated, use the "config.fxp-asset.pattern-skip-version" option
The "extra.asset-vcs-driver-options" option is deprecated, use the "config.fxp-asset.vcs-driver-options" option
```

Does this mean I should switch ASAP, but it's still recognized - or doesn't it have any affect all now?

**Comment by francoispluchino** [comment]
See the section _Dreprecated options_ in the [release page](https://github.com/fxpio/composer-asset-plugin/releases/tag/v1.3.0).

1. The deprecated options are not removed to keep the compatibility
2. All the options in the extra section are functional, and a warning will appear in the console if they are used
3. All the options in the `config.fxp-asset` section will override the options still defined in the `extra` section

**Comment by schmunk42** [comment]
> All the options in the config.fxp-asset section will override the options still defined in the extra section

What if I have defined "old" options in extra, like `vcs-driver-options` and "new" ones in `config` like `git-skip-update`, will they get merged?

**Comment by francoispluchino** [comment]
Only the first key, like the behavior of the `array_merge` function, and not like the `array_merge_recursive` function, so no. 

But if you interested to add this behavior, It's in the method `Fxp\Composer\AssetPlugin\Config\ConfigBuilder::injectDeprecatedConfig()` that it happens.

**Comment by schmunk42** [comment]
It's related to https://github.com/fxpio/composer-asset-plugin/issues/292 - I refactored some `composer.json` files and noticed, that the asset-plugin started to do API requests again, although I've disabled them in the config.

Maybe it would be an option to enforce configuration either via `extra` (with a warning) or via `config` - but if both are specified the plugin would exit with an error. WDYT?
